secret_store/src/listener/service_contract.rs

```diff
@@ -233,6 +298,13 @@ impl Iterator for PendingRequestsIterator {
 	}
 }
 
+/// Get hash of the last block with at least n confirmations.
+fn get_confirmed_block_hash(client: &Client, confirmations: u64) -> Option<H256> {
+	client.block_number(BlockId::Latest)
+		.and_then(|b| b.checked_sub(confirmations))
```

Comment:
It's an edge case, but I would say that for `b < confirmations` we should just return `0`

Reply:
Fixed. Just curious - the only reason is to process requests from block#0 asap (instead of waiting for 3 blocks)? Or is there some other problem with previous pattern?

Reply:
Yeah. I think that getting `None` from this method would indicate some errornous state (non existing historic block?) and it wouldn't be exactyl the same semantics for block `0`.